### ISSUE TITLE
linter-swagger showing error at startup and not working

### ISSUE THREAD

**Issue by ssribeiro** [AtomLinter/linter-swagger#82]
Steps to reproduce: 
> A brand new installed Ubuntu MATE 64bits on Virtual Box to make shure it is not related anything else.
> A brand new installed Atom Text Editor 1.34.0
> Installed Packages: atom-ide-ui, ide-json, linter, linter-swagger

> create a new file named anything.json
> put property "openapi": "3.0.0",

You got: 

```
Uncaught TypeError: Cannot read property 'start' of undefined

/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:34

TypeError: Cannot read property 'start' of undefined
    at Function.t.exports.Range.fromObject (/usr/share/atom/resources/app/static/<embedded>:14:46520)
    at linterMessageV2ToDiagnosticMessage (/home/test/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/atom-ide-diagnostics/lib/services/LinterAdapter.js:233:24)
    at linterMessagesToDiagnosticUpdate (/home/test/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/atom-ide-diagnostics/lib/services/LinterAdapter.js:251:56)
    at a._RxMin.Observable.defer.switchMap.linterMessages [as project] (/home/test/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/atom-ide-diagnostics/lib/services/LinterAdapter.js:325:22)
    at a._next (/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:261:315)
    at a.next (/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:67:406)
    at a.notifyNext (/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:88:375)
    at a._next (/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:90:52)
    at a.next (/home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:67:406)
    at /home/test/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:33:484
    at <anonymous>
```

**Comment by Arcanemagus** [comment]
I'm unable to reproduce this with the following contents in `anything.json`:
```json
{
  "openapi": "3.0.0"
}
```

Could you upload the _exact_ file you are seeing this on?

**Comment by ssribeiro** [comment]
I removed the package "atom-ide-ui"
The file look exactly like yours.
Everthing brand new installed, restarted..
The error message is gone, but no errors are showing up on tab.
Still not working. No lint messages at all.

linter-csslint works just fine perfect. 

I am shure not crazy about that. It is just NOT WORKING.

Brand new installed everthing on VirtualBox:

ubuntu mate 18.04.2
atom 1.36.1

**Comment by Arcanemagus** [comment]
Unless you installed and/or enabled `linter` + `linter-ui-default` after disabling `atom-ide-ui` then you currently have nothing that actually calls this package.

**Comment by broganross** [comment]
I am getting a similar error:

    Uncaught (in promise) TypeError: Cannot read property 'start' of undefined (~/.atom/packages/linter/lib/helpers.js)
      at messageKey (~/.atom/p…r/lib/helpers.js:76)
      at Object.normalizeMessages (~/.atom/p…/lib/helpers.js:112)
      at ~/.atom/p…ter-registry.js:139

I uninstalled `atom-ide-u`, enabled `linter`, `linter-ui-default`, and `linter-swagger`.
I was digging around and it looks like the messages coming out of provideLinter don't follow what is defined [here](https://steelbrain.me/linter/types/linter-message-v2.html).  I haven't done any Atom package development so I could be totally off, but as I started messing with it it looked like changing the messages to match the documentation was helping.

Currently using Atom: 1.40.1, linter: 2.3.1
